source/fuzz/transformation_add_copy_memory.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2020 Vasyl Teliman
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/fuzzer_util.h"
+#include "source/fuzz/instruction_descriptor.h"
+#include "source/fuzz/transformation_add_copy_memory.h"
+#include "source/opt/instruction.h"
+
+namespace spvtools {
+namespace fuzz {
+
+TransformationAddCopyMemory::TransformationAddCopyMemory(
+    const protobufs::TransformationAddCopyMemory& message)
+    : message_(message) {}
+
+TransformationAddCopyMemory::TransformationAddCopyMemory(
+    const protobufs::InstructionDescriptor& instruction_descriptor,
+    uint32_t target_id, uint32_t source_id) {
+  *message_.mutable_instruction_descriptor() = instruction_descriptor;
+  message_.set_target_id(target_id);
+  message_.set_source_id(source_id);
+}
+
+bool TransformationAddCopyMemory::IsApplicable(opt::IRContext* ir_context,
```

Comment:
It's a bit painful, but we should probably walk through the pointee type and check that it does not use OpTypeRuntimeArray anywhere in its structure - the SPIR-V spec says that OpTypeRuntimeArray is not compatible with this instruction.

Reply:
Are there any other types similar to `OpTypeRuntimeArray` (i.e. their size is unknown at compile-time)?

Reply:
Fixed in 5342d07.